diff --git a/app/src/main/java/com/winlator/cmod/XServerDisplayActivity.java b/app/src/main/java/com/winlator/cmod/XServerDisplayActivity.java
--- a/app/src/main/java/com/winlator/cmod/XServerDisplayActivity.java
+++ b/app/src/main/java/com/winlator/cmod/XServerDisplayActivity.java
@@ -20,6 +20,7 @@
+import com.winlator.hud.WinlatorHUD;
@@ -150,6 +151,10 @@
    private void init() {
        setContentView(R.layout.xserver_display_activity);
+        WinlatorHUD.init(this);
    }
@@ -200,6 +205,10 @@
    protected void onDestroy() {
+        WinlatorHUD.release();
        super.onDestroy();
    }
